python/ray/experimental/gpu_object_manager/gpu_object_store.py

```diff
@@ -119,24 +120,32 @@ class GPUObjectStore:
     """
 
     def __init__(self):
-        # A dictionary that maps from an object ID to a list of tensors.
+        # A dictionary that maps from an object ID to a queue of tensor lists.
+        # Each entry in the queue is a list of tensors for one copy of the object.
         #
         # Note: Currently, `_gpu_object_store` is only supported for Ray Actors.
-        self._gpu_object_store: Dict[str, List["torch.Tensor"]] = {}
+        self._gpu_object_store: Dict[str, deque[List["torch.Tensor"]]] = defaultdict(deque)
         # Synchronization for GPU object store.
         self._lock = threading.RLock()
         # Signal when an object becomes present in the object store.
         self._object_present_cv = threading.Condition(self._lock)
         # A set of object IDs that are the primary copy.
         self._primary_gpu_object_ids: Set[str] = set()
 
+        
+
     def has_object(self, obj_id: str) -> bool:
         with self._lock:
-            return obj_id in self._gpu_object_store
+            return obj_id in self._gpu_object_store and len(self._gpu_object_store[obj_id]) > 0
 
     def get_object(self, obj_id: str) -> Optional[List["torch.Tensor"]]:
         with self._lock:
-            return self._gpu_object_store[obj_id]
+            queue = self._gpu_object_store.get(obj_id)
+            if queue and len(queue) > 0:
+                # For primary copies, we peek at the front without removing
+                # For non-primary, we also peek (pop happens separately)
+                return queue[0]
+            return None
```

Comment:
Let's keep the same behavior as before where we would error if the key was not found.